WordPress/gutenberg
# adds simple logic that keeps authored captions in galleries

## Description
Attempted fix for #8310 

## How has this been tested?
- add a gallery to a post
- edit captions of images
- edit the gallery
- the captions should remain
- delete a caption or more
- edit the gallery and add captions to the files which dont have captions
- the images with empty captions should get the caption from the gallery

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
